ui/app/controllers/settings/tokens.js

```diff
@@ -120,25 +127,43 @@ export default class Tokens extends Controller {
   }
 
   async validateSSO() {
+    let redirectURL;
+    if (Ember.testing) {
+      redirectURL = this.router.currentURL;
+    } else {
+      redirectURL = new URL(window.location.toString());
+      redirectURL.search = '';
+      redirectURL = redirectURL.href;
+    }
+
     const res = await this.token.authorizedRequest(
       '/v1/acl/oidc/complete-auth',
       {
         method: 'POST',
         body: JSON.stringify({
-          AuthMethod: window.localStorage.getItem('nomadOIDCAuthMethod'),
+          AuthMethodName: window.localStorage.getItem('nomadOIDCAuthMethod'),
           ClientNonce: window.localStorage.getItem('nomadOIDCNonce'),
           Code: this.code,
           State: this.state,
+          RedirectURI: redirectURL,
         }),
       }
     );
 
     if (res.ok) {
       const data = await res.json();
-      this.token.set('secret', data.ACLToken);
-      this.verifyToken();
+      this.clearTokenProperties();
+      this.token.set('secret', data.SecretID);
       this.state = null;
       this.code = null;
+
+      this.resetStore();
```

Comment:
So we're unloading the store on line 155, then editing a token (from an empty store), and then unloading the store, again here. And then we fetch the token on line 163.

Maybe we might be doing too much work here. There's a small chance that once the store has a large number of items that we'll run into challenges with asynchronocity.

Should we get an integration test on this?

Reply:
You may be correct here! This looks like it could be a redundant unload of the store. 

FWIW, we do run this in a regular acceptance test [here](https://github.com/hashicorp/nomad/blob/c314661baff456c7ea04ff531b1c2b9f5a4b2150/ui/tests/acceptance/token-test.js#L187) with a non-trivial amount of store load.

Reply:
Interesting! There's a `forEach` being called when you [inspect](https://github.com/emberjs/data/blob/v4.7.3/packages/store/addon/-private/store-service.ts#L650) `InstanceCache` which is called on `unloadAll`. But, I'm not complaining! Less Work!!!